### DISCUSSION TITLE
Server and client diffrent network with same public IP range won't connect.

### DISCUSSION THREAD

**Discussion by Nawaki89** [rustdesk/rustdesk-server-pro#291] [Q&A]
Hello! I'v setup a server on its own network and public IP small server setup env.
we use the same fiber connection as the branch in my city.  I setup the server pro and generated some custom clients installed remote in diffrent branch offices and it connects and works splendid.   but when i install the client for the branch that is using the same public ip range as the server i get "Not ready, check your connection" 

anyone encounter similar issue and have any tips?.

**Comment by Nawaki89** [comment]
hmm some might apply but its 2 diffrent routers, 2 diffrent networks its just the same public IP range.  they do share the same DNS servers.

**Reply by rustdesk** [reply]
`telnet <rustdesk server ip> 21116` work?

**Reply by Nawaki89** [reply]
if I goes in to the "branch" network and try telnet < public ip > 21116 it failes to connect.
if I try from home with telnet < public IP > 21116 I get a respons with alot of random symbols.

**Reply by rustdesk** [reply]
So this is totally networking problem, need to ask some networking expert.

**Reply by Nawaki89** [reply]
thank you for pointing me on the right path. I will investigate!

**Reply by Nawaki89** [reply]
Just wanted to give an update even tho it wasn’t a Rust problem but you where nice to point me in the right direction.

the ISP at site even tho I used two different “lines” and built two separate networks with it own public IP.  The office side couldn’t connect to the rust server.   The problem was networking as mentioned.   The ISP have an gateway that manage all the public IP addresses and they had port isolation enable which is standard procedure but that blocked the office to utilize RustDesk program since the CPE as they call it didn’t allow the interface to communicate even tho the traffic was routed with public IP.   ISP disable that feature on the concerning ports and it jump online straight away. 

If someone els encounter similar problems worth taking a look, fiber connections commonly are behind some kind of managed unit for business use.  

Thanks for the help getting me in the right direction.

**Reply by rustdesk** [reply]
Apologize for our poor knowledge of networking.  Maybe proxy can help you.

![image](https://github.com/rustdesk/rustdesk-server-pro/assets/71636191/93c472c6-b84c-409e-bb74-be9c7c164d70)

**Reply by Nawaki89** [reply]
I’m no expert either but I managed to decipher the problem after you pointed me to the right area so thanks!

proxy might have worked but I solved it in the hardware instead felt more reliable :)
